question-servers/freeform.js

```diff
@@ -6,10 +6,74 @@ var path = require('path');
 var mustache = require('mustache');
 var cheerio = require('cheerio');
 
-var elements = require('./elements');
+var {
+    elements,
+    dependencies: elementDependencies
+} = require('./elements');
 var codeCaller = require('../lib/code-caller');
 
 module.exports = {
+    /**
+     * Computes the dependencies given a Set of element names.
+     * @param  {Set}      elements
+     * @param  {Function} callback Will be called with an array of <link>/<script> tags
+     */
+    getElementDependencies: function(elements, callback) {
+        const elementStyleFiles = [];
+        const elementScriptFiles = [];
+        for (const elem of elements) {
+            elementStyleFiles.push(`${elem}.css`);
+            elementScriptFiles.push(`${elem}.js`);
+        }
+        const styleUrls = new Set();
+        const scriptUrls = new Set();
+        async.parallel([
+            (callback) => {
+                async.each(elementStyleFiles, (file, callback) => {
+                    fs.access(path.join(__dirname, 'elements', file), (err) => {
+                        if (!err) {
+                            styleUrls.add(`/pl/static/elements/${file}`);
+                            //headerHtmls.push(`<link href="/pl/static/elements/${file}" rel="stylesheet" />`);
+                        }
+                        callback(null);
+                    });
+                }, () => callback(null));
+            },
+            (callback) => {
+                async.each(elementScriptFiles, (file, callback) => {
+                    fs.access(path.join(__dirname, 'elements', file), (err) => {
+                        if (!err) {
+                            scriptUrls.add(`/pl/static/elements/${file}`);
+                            //headerHtmls.push(`<script type="text/javascript" src="/pl/static/elements/${file}"></script>`);
+                        }
+                        callback(null);
+                    });
+                }, () => callback(null));
+            }
+        ], () => {
+            // Now that we've ~automagically~ added all available files, let's
+            // add any other declared dependencies
+            const globalScriptUrls = new Set();
+            for (const elem of elements) {
+                const deps = elementDependencies[elem];
+                if (!deps) continue;
+                deps.styles && deps.styles.forEach((file) => styleUrls.add(`/pl/static/elements/${file}`));
```

Comment:
I do agree with you that someone will screw this up and list things in the wrong order, but presumably they'll find out immediately when backbone can't find lodash, for example?